### PR TITLE
CBR-2214: Invoke share portfolio stepfunction within Unmanaged account onboarding

### PR DESCRIPTION
**Pull request by jophine-antony**
Invoke share portfolio stepfunction within Unmanaged account onboarding
Modified:
  - template.yaml
  - index_create_aws_iam_roles.js

### CODE REVIEW

File: src/json/state_machine_input.json [hunk 1]
```diff
@@ -142,6 +142,7 @@
     },
     "result": null
   },
+  "share_portfolio_params": null,
```
Comment (SonalOjha) [inline src/json/state_machine_input.json]: Define this as 

{
          aws_region_id: null,
          aws_account_id: null
          role_details: {
            role_name: null,
            external_id: null
          }
}

File: src/index_create_aws_iam_roles.js [hunk 2]
```diff
@@ -62,6 +62,17 @@ exports.handler = function(event, context, callback) {
         payload.assumeRolePolicyDocument.Statement[0].Principal.AWS = `arn:aws:iam::${process
           .env.DATADOG_AWD_ID}:root`;
       }
+      // Collect share portfolio params data for stepfunction invocation
+      if (payload.roleName == 'sgas_sc_admin') {
```
[the remark this answers is not in the document]
Comment (jophine-antony) [reply src/index_create_aws_iam_roles.js]: Addressed comments

File: src/index_create_aws_iam_roles.js [hunk 3]
```diff
@@ -64,14 +64,10 @@ exports.handler = function(event, context, callback) {
...
       if (payload.roleName == 'sgas_sc_admin') {
-        event.share_portfolio_params = {
-          aws_region_id: 'us-east-1',
-          aws_account_id: payload.account,
-          role_details: {
-            role_name: payload.roleName,
-            external_id: payload.externalId
-          }
-        }
+        event.share_portfolio_params.aws_region_id = 'us-east-1';
```
Comment (SonalOjha) [inline src/index_create_aws_iam_roles.js]: Do we need this to be hard coded ?

**Comment by SonalOjha** [comment]
This PR is yet to be worked on and so requesting to create another once the changes and testing is completed

**Pull request closed without merging**